### PR TITLE
Fixed some warnings.

### PR DESCRIPTION
**Pull request by mpimenov**

### CODE REVIEW

File: coding/coding_tests/dd_vector_test.cpp [hunk 1]
```diff
@@ -19,7 +19,7 @@ UNIT_TEST(DDVector_Smoke)
   TEST_EQUAL(2, v[1], ());
   TEST_EQUAL(3, v[2], ());
   Vector::const_iterator it = v.begin();
-  for (size_t i = 0; i < v.size(); ++i, ++it)
+  for (Vector::size_type i = 0; i < v.size(); ++i, ++it)
```
Comment (ygorshenin) [inline coding/coding_tests/dd_vector_test.cpp]:
```c++
for (auto const value : v)
  TEST_EQUAL(v, *it++, ());
```

File: routing/routing_tests/osrm_router_test.cpp [hunk 2]
```diff
@@ -117,7 +118,7 @@ void TestMapping(InputDataT const & data,
 
     for (size_t i = 0; i < mapping.GetSegmentsCount(); ++i)
     {
-      TOsrmNodeId const node = mapping.GetNodeId(i);
+      TOsrmNodeId const node = mapping.GetNodeId(base::asserted_cast<uint32_t>(i));
```
Comment (ygorshenin) [inline routing/routing_tests/osrm_router_test.cpp]: In tests checked_cast is better.

File: routing/routing_tests/nearest_edge_finder_tests.cpp [hunk 3]
```diff
@@ -16,7 +18,7 @@ void TestNearestOnMock1(m2::PointD const & point, size_t const candidatesCount,
   NearestEdgeFinder finder(point);
   for (size_t i = 0; i < graph->GetRoadCount(); ++i)
   {
-    FeatureID const featureId = MakeTestFeatureID(i);
+    FeatureID const featureId = MakeTestFeatureID(base::asserted_cast<uint32_t>(i));
```
Comment (ygorshenin) [inline routing/routing_tests/nearest_edge_finder_tests.cpp]: Please, use checked_cast in tests.

File: routing/routing_tests/road_graph_builder.cpp [hunk 4]
```diff
@@ -88,7 +88,7 @@ double RoadGraphMockSource::GetMaxSpeedKMPH() const
 void RoadGraphMockSource::ForEachFeatureClosestToCross(m2::PointD const & /* cross */,
                                                        ICrossEdgesLoader & edgesLoader) const
 {
-  for (size_t roadId = 0; roadId < m_roads.size(); ++roadId)
+  for (uint32_t roadId = 0; roadId < m_roads.size(); ++roadId)
```
Comment (ygorshenin) [inline routing/routing_tests/road_graph_builder.cpp]: Why? m_roads.size() is size_t, why not  to cast size_t to uint32_t when you call MakeTestFeatureId?

File: search/locality_scorer.cpp [hunk 5]
```diff
@@ -62,7 +64,8 @@ void LocalityScorer::GetTopLocalities(MwmSet::MwmId const & countryId, BaseConte
       // Skip locality candidates that match only numbers.
       if (!m_params.IsNumberTokens(tokenRange))
       {
-        intersection.ForEach([&](uint32_t featureId) {
+        intersection.ForEach([&](uint64_t bit) {
+          uint32_t const featureId = base::asserted_cast<uint32_t>(bit);
```
Comment (ygorshenin) [inline search/locality_scorer.cpp]: auto const

File: search/hotels_filter.cpp [hunk 6]
```diff
@@ -111,7 +112,8 @@ HotelsFilter::Descriptions const & HotelsFilter::GetDescriptions(MwmContext cons
 
   auto const hotels = m_hotels.Get(context);
   auto & descriptions = m_descriptions[mwmId];
-  hotels.ForEach([&descriptions, &context](uint32_t id) {
+  hotels.ForEach([&descriptions, &context](uint64_t bit) {
+    uint32_t const id = base::asserted_cast<uint32_t>(bit);
```
Comment (ygorshenin) [inline search/hotels_filter.cpp]: auto const

File: coding/coding_tests/dd_vector_test.cpp (hunk 1 above)
Comment (mpimenov) [inline coding/coding_tests/dd_vector_test.cpp]: done

File: routing/routing_tests/osrm_router_test.cpp (hunk 2 above)
Comment (mpimenov) [inline routing/routing_tests/osrm_router_test.cpp]: done

File: routing/routing_tests/nearest_edge_finder_tests.cpp (hunk 3 above)
Comment (mpimenov) [inline routing/routing_tests/nearest_edge_finder_tests.cpp]: done

File: routing/routing_tests/road_graph_builder.cpp (hunk 4 above)
Comment (mpimenov) [inline routing/routing_tests/road_graph_builder.cpp]: That was my first version but I thought that an implicit cast is good enough.
done

File: search/locality_scorer.cpp (hunk 5 above)
Comment (mpimenov) [inline search/locality_scorer.cpp]: done

File: search/hotels_filter.cpp (hunk 6 above)
Comment (mpimenov) [inline search/hotels_filter.cpp]: done
here and in geocoder.cpp

**Comment by mpimenov** [comment]
PTAL @ygorshenin

**Pull request merged**